pyrobosim/pyrobosim/core/yaml_utils.py

```diff
@@ -75,7 +76,17 @@ def create_world(self):
                 )
             else:
                 obj_data = None
-            self.world.set_metadata(locations=loc_data, objects=obj_data)
+
+            loc_data = [loc_data] if isinstance(loc_data, str) else loc_data or []
+            obj_data = [obj_data] if isinstance(obj_data, str) else obj_data or []
+
+            self.world.set_metadata(locations=loc_data[0], objects=obj_data[0])
+
+            if len(loc_data) > 1 and len(obj_data) > 1:
+                for location, object in zip_longest(
+                    loc_data[1:], obj_data[1:], fillvalue=None
+                ):
+                    self.world.add_metadata(locations=location, objects=object)
```

Comment:
Same as my comment on the demo -- can we just call `add_metadata` in all cases so we don't have to separate the 0th index from others?

Also, I'm pretty sure this will error if `loc_data` or `obj_data` is empty (size 0).

Lastly, instead of using `zip_longest`, consider my earliest suggestion where `World.add_metadata()` and `World.set_metadata()` can just accept lists of metadata in addition to single items.